cli/cmd/shell.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/microsoft/retina/shell"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	configFlags              *genericclioptions.ConfigFlags
+	matchVersionFlags        *cmdutil.MatchVersionFlags
+	retinaShellImageRepo     string
+	retinaShellImageVersion  string
+	mountHostFilesystem      bool
+	allowHostFilesystemWrite bool
+	hostPID                  bool
+	capabilities             []string
+	timeout                  time.Duration
+)
+
+const defaultRetinaShellImageRepo = "ghcr.io/microsoft/retina/retina-shell"
```

Comment:
Few questions about this part:

* I'd prefer to use MCR instead of ghcr, since AKS clusters are required to have access to MCR. Everything else in this repo seems to be using ghcr, but is it possible to use MCR instead? (can make sure the image is replicated there before merging this).
* Defaulting to the Retina CLI version for the image tag. This works when the version is set in release builds, but will fail otherwise since we're defaulting to 0.0.5 here https://github.com/microsoft/retina/blob/2b109a35e45e5bc8e6c32a1cdb3578dacddc772e/cli/cmd/version.go#L12-L14 could we bump this later to a version that includes retina shell? (once it exists)

Reply:
You could make it a var, have it set by default to GHCR and then patch this var at link time with MCR if needed.

Reply:
yeah, that's an option. Do we do any patching like that today when building the retina kubectl plugin? (Ideally the official plugin release would use MCR so it will "just work" in any AKS cluster)

Reply:
`Version` got moved to buildinfo and the default fallback removed. I updated this PR to: (1) use MCR by default, and (2) use buildinfo.Version and check that it's actually set (may not be when built locally).